Source/SwiftLintFramework/Models/Linter.swift

```diff
@@ -108,59 +109,101 @@ private extension Rule {
 }
 
 public struct Linter {
+    public let file: File
+    fileprivate let rules: [Rule]
+    fileprivate let cache: LinterCache?
+    fileprivate let configuration: Configuration
+    fileprivate let compilerArguments: [String]
+
+    public init(file: File, configuration: Configuration = Configuration()!, cache: LinterCache? = nil,
+                compilerArguments: [String] = []) {
+        self.file = file
+        self.cache = cache
+        self.configuration = configuration
+        self.compilerArguments = compilerArguments
+        let rules = configuration.rules.filter { rule in
+            if compilerArguments.isEmpty {
+                return !(rule is AnalyzerRule)
+            } else {
+                return rule is AnalyzerRule
+            }
+        }
+        self.rules = rules
+    }
+
+    public func collect(into storage: RuleStorage) -> CollectedLinter {
+        DispatchQueue.concurrentPerform(iterations: rules.count) { idx in
+            rules[idx].collect(infoFor: file, into: storage, compilerArguments: compilerArguments)
+        }
+        return CollectedLinter(from: self)
+    }
+}
+
+public struct CollectedLinter {
```

Comment:
What do you think of calling this a "CollectingLinter" instead?

Reply:
Hmm, I'm not married to this naming convention, but I think that calling this type a `CollectingLinter` would imply that _it_ does the collecting, when in actually `Linter` is what runs collection. Linter and CollectedLinter are related by the `collect` function, i.e.

```swift
Linter.collect(into storage: RuleStorage) -> CollectedLinter
```

What if we renamed Linter to `CollectingLinter` and CollectedLinter to `Linter`? Then the relation would be
```swift
CollectingLinter.collect(into storage: RuleStorage) -> Linter
```

Or we could stop insisting that both types are a "linter", and name them something like
```swift
Collecter.collect(into storage: RuleStorage) -> Linter
```

Reply:
Oh I see. I think "Collected Linter" makes sense, but this could use a docstring to explain that.